modules/subchannel/src/meshgenerators/QuadSubChannelMeshGenerator.C

```diff
@@ -0,0 +1,502 @@
+  unsigned int node_id = 0;
+  for (unsigned int iy = 0; iy < _ny; iy++)
+  {
+    for (unsigned int ix = 0; ix < _nx; ix++)
+    {
+      int i_ch = _nx * iy + ix;
+      _nodes[i_ch].reserve(_n_cells);
+      for (unsigned int iz = 0; iz < _n_cells + 1; iz++)
+      {
+        _nodes[i_ch].push_back(mesh_base->add_point(
+            Point(_pitch * ix - offset_x, _pitch * iy - offset_y, _z_grid[iz]), node_id++));
+      }
+    }
+  }
+
+  // Add the elements which in this case are 2-node edges that link each
+  // subchannel's nodes vertically.
+  unsigned int elem_id = 0;
+  for (unsigned int iy = 0; iy < _ny; iy++)
+  {
+    for (unsigned int ix = 0; ix < _nx; ix++)
+    {
+      for (unsigned int iz = 0; iz < _n_cells; iz++)
+      {
+        Elem * elem = new Edge2;
+        elem->subdomain_id() = _block_id;
+        elem->set_id(elem_id++);
+        elem = mesh_base->add_elem(elem);
+        const int indx1 = ((_n_cells + 1) * _nx) * iy + (_n_cells + 1) * ix + iz;
+        const int indx2 = ((_n_cells + 1) * _nx) * iy + (_n_cells + 1) * ix + (iz + 1);
+        elem->set_node(0) = mesh_base->node_ptr(indx1);
+        elem->set_node(1) = mesh_base->node_ptr(indx2);
+
+        if (iz == 0)
+          boundary_info.add_side(elem, 0, 0);
+        if (iz == _n_cells - 1)
+          boundary_info.add_side(elem, 1, 1);
+      }
+    }
+  }
+
+  boundary_info.sideset_name(0) = "inlet";
+  boundary_info.sideset_name(1) = "outlet";
+  boundary_info.nodeset_name(0) = "inlet";
+  boundary_info.nodeset_name(1) = "outlet";
+  mesh_base->subdomain_name(_block_id) = name();
+  mesh_base->prepare_for_use();
+
+  // move the meta data into QuadSubChannelMesh
+  auto & sch_mesh = static_cast<QuadSubChannelMesh &>(*_mesh);
+  sch_mesh._unheated_length_entry = _unheated_length_entry;
+  sch_mesh._heated_length = _heated_length;
+  sch_mesh._unheated_length_exit = _unheated_length_exit;
+  sch_mesh._z_grid = _z_grid;
+  sch_mesh._k_grid = _k_grid;
+  sch_mesh._spacer_z = _spacer_z;
+  sch_mesh._spacer_k = _spacer_k;
+  sch_mesh._z_blockage = _z_blockage;
+  sch_mesh._index_blockage = _index_blockage;
+  sch_mesh._reduction_blockage = _reduction_blockage;
+  sch_mesh._kij = _kij;
+  sch_mesh._pitch = _pitch;
+  sch_mesh._pin_diameter = _pin_diameter;
+  sch_mesh._n_cells = _n_cells;
+  sch_mesh._nx = _nx;
+  sch_mesh._ny = _ny;
+  sch_mesh._n_channels = _n_channels;
+  sch_mesh._n_gaps = _n_gaps;
+  sch_mesh._n_pins = _n_pins;
+  sch_mesh._gap = _gap;
+  sch_mesh._nodes = _nodes;
+  sch_mesh._gapnodes = _gapnodes;
+  sch_mesh._gap_to_chan_map = _gap_to_chan_map;
+  sch_mesh._gap_to_pin_map = _gap_to_pin_map;
+  sch_mesh._chan_to_gap_map = _chan_to_gap_map;
+  sch_mesh._chan_to_pin_map = _chan_to_pin_map;
+  sch_mesh._pin_to_chan_map = _pin_to_chan_map;
+  sch_mesh._sign_id_crossflow_map = _sign_id_crossflow_map;
+  sch_mesh._gij_map = _gij_map;
+  sch_mesh._subchannel_position = _subchannel_position;
+  sch_mesh._subch_type = _subch_type;
```

Comment:
umm if you dont have all these setters, maybe keep friend here for the sake of time